com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl

```diff
@@ -896,164 +129,312 @@ half3 VertexLighting(float3 positionWS, half3 normalWS)
+    lightingData.giColor = GlobalIllumination(brdfData, brdfDataClearCoat, surfaceData.clearCoatMask,
+                                              inputData.bakedGI, aoFactor.indirectAmbientOcclusion,
+                                              inputData.normalWS, inputData.viewDirectionWS);
 
     if (IsMatchingLightLayer(mainLight.layerMask, meshRenderingLayers))
     {
-        color += LightingPhysicallyBased(brdfData, brdfDataClearCoat,
-                                         mainLight,
-                                         inputData.normalWS, inputData.viewDirectionWS,
-                                         surfaceData.clearCoatMask, specularHighlightsOff);
+        lightingData.mainLightColor = LightingPhysicallyBased(brdfData, brdfDataClearCoat,
+                                                              mainLight,
+                                                              inputData.normalWS, inputData.viewDirectionWS,
+                                                              surfaceData.clearCoatMask, specularHighlightsOff);
     }
 
-#ifdef _ADDITIONAL_LIGHTS
+    #if defined(_ADDITIONAL_LIGHTS)
     uint pixelLightCount = GetAdditionalLightsCount();
+
     LIGHT_LOOP_BEGIN(pixelLightCount)
-        Light light = GetAdditionalLight(lightIndex, inputData.positionWS, shadowMask);
+        Light light = GetAdditionalLight(lightIndex, inputData, shadowMask, aoFactor);
 
         if (IsMatchingLightLayer(light.layerMask, meshRenderingLayers))
         {
-            #if defined(_SCREEN_SPACE_OCCLUSION) && !defined(_SURFACE_TYPE_TRANSPARENT)
-                light.color *= aoFactor.directAmbientOcclusion;
-            #endif
-            color += LightingPhysicallyBased(brdfData, brdfDataClearCoat,
-                                             light,
-                                             inputData.normalWS, inputData.viewDirectionWS,
-                                             surfaceData.clearCoatMask, specularHighlightsOff);
+            lightingData.additionalLightsColor += LightingPhysicallyBased(brdfData, brdfDataClearCoat, light,
+                                                                          inputData.normalWS, inputData.viewDirectionWS,
+                                                                          surfaceData.clearCoatMask, specularHighlightsOff);
         }
     LIGHT_LOOP_END
-#endif
-
-#ifdef _ADDITIONAL_LIGHTS_VERTEX
-    color += inputData.vertexLighting * brdfData.diffuse;
-#endif
+    #endif
 
-    color += surfaceData.emission;
+    #if defined(_ADDITIONAL_LIGHTS_VERTEX)
+    lightingData.vertexLightingColor += inputData.vertexLighting * brdfData.diffuse;
+    #endif
 
-    return half4(color, surfaceData.alpha);
+    return CalculateFinalColor(lightingData, surfaceData.alpha);
 }
 
+// Deprecated: Use the version which takes "SurfaceData" instead of passing all of these arguments...
```

Comment:
I think we could move all of the deprecated backwards compatibility functions into a separate file. Maybe lighting.deprecated.hlsl